File: algorithms/selection_sort/selection_sort.js
// Time O(1) space O(n^2)
function selectionSort(array) {
  let currentIndex = 0;
  while (currentIndex < array.length - 1) {
    let smallIndex = currentIndex;
    for (let i = currentIndex; i < array.length + 1; i++) {
      if (array[smallIndex] > array[i]) {
        smallIndex = i;
      }
    }
    let tmp = array[smallIndex];
    array[smallIndex] = array[currentIndex];
    array[currentIndex] = tmp;
    currentIndex += 1;
  }
  return array;
}

console.log(selectionSort([24, 26, 47, 2, 5, 6, 8, 9, 12, 15, 20]));

File: algorithms/selection_sort/selection_sort.py
# Time O(1) space O(n^2)
def selectionSort(array):
    currentIndex = 0
    while currentIndex < len(array) - 1:
        smallIndex = currentIndex
        for i in range(currentIndex + 1, len(array)):
            if array[smallIndex] > array[i]:
                smallIndex = i
        array[currentIndex], array[smallIndex] = array[smallIndex], array[currentIndex]
        currentIndex += 1
    return array

print(selectionSort([24, 26, 47, 2, 5, 6, 8, 9, 12, 15, 20]))
